src/types/safe.rs

```diff
@@ -19,28 +20,23 @@ impl Safe {
     }
     /// @returns how much of their own tokens a user can send to receiver.
     pub fn trust_transfer_limit(&self, receiver: &Safe, trust_percentage: u8) -> U256 {
-        if receiver.organization {
-            // TODO treat this as "return to owner"
-            // i.e. limited / only constrained by the balance edge.
-            self.balance(&self.token_address)
+        if receiver.organization || receiver.token_address == self.token_address {
+            return self.balance(&self.token_address);
         } else {
             let receiver_balance = receiver.balance(&self.token_address);
+            let one_hundred = U256::from(100_u128);
 
-            let amount = (receiver.balance(&receiver.token_address)
+            let max = (receiver.balance(&receiver.token_address)
                 * U256::from(trust_percentage as u128))
-                / U256::from(100);
-            let scaled_receiver_balance =
-                receiver_balance * U256::from((100 - trust_percentage) as u128) / U256::from(100);
-            if amount < receiver_balance {
-                U256::from(0)
-            } else {
-                // TODO it should not be "min" - the second constraint
-                // is set by the balance edge.
-                min(
-                    amount - scaled_receiver_balance,
-                    self.balance(&self.token_address),
-                )
+                / one_hundred;
+
+            if max < receiver_balance {
+                return U256::from(0);
             }
+
+            let receiver_balance_scaled = receiver_balance * (one_hundred - U256::from(trust_percentage as u128)) / one_hundred;
```

Comment:
Have you checked this issue?
We discovered that there is a potencial issue in the hub contract.
https://github.com/CirclesUBI/circles-contracts/issues/173